### PR TITLE
revise pkg-tests.sh

### LINKED ISSUE #1: Improve `pkg-tests.sh` to print the GAP output immediately, and not first into a log file
Right now, `pkg-tests.sh` runs `make testpackages`, which logs all output into files. We only print those files later. But that means that in the case of test runs which time out (e.g. due to an endless recursion/loop), we see nothing, just the information by Travis that there was a timeout. In particular, we don't see where it hangs (in which of the three different test suite runs, and in which file and line)

It would IMHO be much nicer if each test was run and immediately printed its output, while running. That would also mean that one gets output soon for the first of the three runs through the test suite; i.e., I could already see that the first of three test suite runs passed/failed, before the other two are done (granted, that's not something very important, it's just another bonus of such a change).

Technically, there are several ways to do this. We could modify `testpackages`, but my understanding is that @alex-konovalov uses this elsewhere.

The easiest would be IMHO to just not use `testpackages` anymore, and just run the tests directly, it's not hard after all. This is what the Makefile rule for it: (UPDATE: accidentally had quoted `testpackages` target here, now corrected)
```
testpackage: all
	$(MKDIR_P) dev/log
	( echo 'SetAssertionLevel( 2 ); ReadGapRoot( "tst/testutil.g" ); \
            SaveWorkspace( "wsp.g" );' | $(TESTGAP) )
	( echo 'CreatePackageTestsInput( "testpackage.in", \
            "dev/log/testpackage1", \
            "$(TESTGAP) -L wsp.g", "false", "$(PKGNAME)" );'\
            | $(TESTGAP) -L wsp.g > /dev/null )
	( chmod 777 testpackage.in; ./testpackage.in; rm testpackage.in )
	( rm wsp.g )
	( echo 'SetAssertionLevel( 2 ); ReadGapRoot( "tst/testutil.g" ); \
            SaveWorkspace( "wsp.g" );' | $(TESTGAPauto) )
	( echo 'CreatePackageTestsInput( "testpackage.in", \
            "dev/log/testpackageA", \
            "$(TESTGAPauto) -L wsp.g", "auto", "$(PKGNAME)" );'\
            | $(TESTGAPauto) -L wsp.g > /dev/null )
	( chmod 777 testpackage.in; ./testpackage.in; rm testpackage.in )
	( rm wsp.g )
	( echo 'SetAssertionLevel( 2 ); ReadGapRoot( "tst/testutil.g" ); \
            SaveWorkspace( "wsp.g" );' | $(TESTGAP) )
	( echo 'CreatePackageTestsInput( "testpackage.in", \
            "dev/log/testpackage2", \
            "$(TESTGAP) -L wsp.g", "true", "$(PKGNAME)" );'\
            | $(TESTGAP) -L wsp.g > /dev/null )
	( chmod 777 testpackage.in; ./testpackage.in; rm testpackage.in )
	( rm wsp.g )
```

Since we only test a single package each time, we don't need to bother with creating workspaces. I also see no benefit in using `CreatePackageTestsInput` (but I may be missing something?). 

Instead, we could use `TestOnePackage ` from <https://github.com/gap-infra/gap-docker-pkg-tests-master-devel/blob/master/ci.g>  (and indeed, perhaps merge that function back into GAP's `TestPackage`). And then pipe a program like this into GAP:
```gap
result:=TestOnePackage("$PKG_NAME");
if result = true then FORCE_QUIT_GAP(0); fi;
FORCE_QUIT_GAP(1);
```
repeated three times with different args for GAP.

### LINKED ISSUE #6: Several failing packages don't seem to run tests at all; end with `cat: 'dev/log/testpackage1*.corelg': No such file or directory`
See e.g. [this log](https://github.com/gap-infra/integration/runs/5294942688?check_suite_focus=true) which ends with
```
cat: 'dev/log/testpackage1*.corelg': No such file or directory
```
Presumably because `make testpackage PKGNAME=$PKG_NAME` is used (see also #1 -- we shouldn't do this anymore) and fails to produce something. We can't see why, though. Perhaps some dependency is missing / was not compiled?

I suspect this will become easy to deal with once we resolved #1. Seems I always circle back to that :-)

### PR DESCRIPTION
**Pull request by fingolfin**
Resolves #1 
Resolves #6

### CODE REVIEW

File: pkg-tests.sh [hunk 1]
```diff
@@ -4,100 +4,10 @@
 
 set -e
 
+# TODO: why do we set TERM here? Perhaps due to https://github.com/gap-system/gap/issues/1617 ?
+# -> Ask Alex and Thomas about this, and either remove it, or else keep it but then document it
```
Comment (wilfwilson) [inline pkg-tests.sh:8]: This was present in the `.travis.yml` files of the old `gap-infra` repositories, it was set before calling this script. I moved the declaration of the variable here, so I'm the one who put it here. But I don't know what it does or if it is still necessary.

**Comment by fingolfin** [comment]
One thing this reveals is that a bunch of package tests "fail" because... there are no package tests, i.e., no `TestFile` in `PackageInfo.g`. E.g. this is the case for `corelg` (although I just made a new release correcting this) or also `Browse`, `Gauss`, `GAPDoc`, `modules`, ...

On the up side, some failing tests are already fixed, e.g. `ace` has a newer release fixing the failure reported here

**Comment by fingolfin** [comment]
I am running tests with this PR now on https://github.com/fingolfin/integration/runs/5312038408?check_suite_focus=true

**Review by wilfwilson — APPROVED** [review]

File: pkg-tests.sh (hunk 1 above)
Comment (olexandr-konovalov) [reply pkg-tests.sh:8]: yes, https://github.com/gap-system/gap/issues/1617 seems relevant, perhaps similar issue was on Travis.

**Pull request merged**